### PR TITLE
Problem submission - Learning gradient descent with synthetic objectives

### PR DESCRIPTION
**Pull request by cjratcliff**

### CODE REVIEW

File: projects/learning-gradient-descent-with-synthetic-objectives.md [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+Date: November 2016 
+Category: Fundamental Research 
+Mailing list: https://groups.google.com/forum/#!forum/aion-learning-gradient-descent-with-synthetic-objectives
+Contact: Chris Ratcliff - c.j.ratcliff@gmail.com
+
+
+## Problem description
+Current optimization algorithms for neural networks such as SGD, RMSProp and Adam are hand-crafted and generally quite simple. This can be partly explained by the high-dimensional, non-convex nature of neural network's objective functions which human intuition, normally limited to three spatial dimensions, is not well-suited for. A learning algorithm, therefore, may be able to design a superior optimizer.
+
+
```
Comment (farizrahman4u) [inline projects/learning-gradient-descent-with-synthetic-objectives.md]: Unnecessary white space.

File: projects/learning-gradient-descent-with-synthetic-objectives.md [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+## How to measure success
+A graph of training loss against the number of iterations for a network trained under different algorithms is commonly used to see which algorithm is better by simple visual inspection. One may also consider plotting loss against time rather than the number of iterations. This would be a harder metric, given the expense of computing one step of an LSTM compared to a single scalar multiplication, as in standard SGD.
+
+
+## Project status
+A formula for generating synthetic objective functions has been created. These functions are differentiable and their dimensionality and degree of non-linearity can be controlled with hyperparameters. 
+
+
```
Comment (farizrahman4u) [inline projects/learning-gradient-descent-with-synthetic-objectives.md]: Here too.

**Review by farizrahman4u — CHANGES_REQUESTED** [review]
2 new lines after each section; Only 1 required between paragraphs of the same section.

**Review by farizrahman4u — APPROVED** [review]

**Comment by fchollet** [comment]
I would like to see more people reviewing this proposal and giving feedback.

**Comment by farizrahman4u** [comment]
@cjratcliff  Do you have links for the stuff mentioned under project status? For the optimizers trained with supervision and reinforcement learning?

**Comment by cjratcliff** [comment]
@farizrahman4u Sure. I've now uploaded the code to Github [here](https://github.com/cjratcliff/optimization). Note that the learned optimizers can't be used as seamlessly as TensorFlow's inbuilt ones right now. Gradients have to be explicitly calculated and passed into the optimizer, which outputs the updates.

Don't expect anything much from the RL version. It's far enough from being complete that it's no better than choosing random updates.
